pkg/controller/data_download_controller.go

```diff
@@ -416,6 +429,29 @@ func (r *DataDownloadReconciler) findSnapshotRestoreForPod(podObj client.Object)
 	return requests
 }
 
+func (r *DataDownloadReconciler) FindDataDownloads(ctx context.Context, cli client.Client, ns string) ([]*velerov2alpha1api.DataDownload, error) {
+	pods := &v1.PodList{}
+	var dataDownloads []*velerov2alpha1api.DataDownload
+	if err := cli.List(ctx, pods, &client.ListOptions{Namespace: ns}); err != nil {
+		r.logger.WithError(errors.WithStack(err)).Error("failed to list pods on current node")
+		return nil, errors.Wrapf(err, "failed to list pods on current node")
+	}
+
+	for _, pod := range pods.Items {
+		if pod.Spec.NodeName != r.nodeName {
```

Comment:
This if judgement should never go true, right? If so, let's add a comment and error log

Reply:
it could get into this condition and it's not one exception, I'll add one debug log

Reply:
Add